### PR TITLE
Add the documentation for the CheckboxToggleNotice UI component

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this pull request

This pull request (PR) adds the documentation for the CheckboxToggleNotice UI component.

## Affected DevDocs pages

It is a new page.

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.2/app/code/Magento/Ui/view/base/web/js/form/element/single-checkbox-toggle-notice.js
-  https://github.com/magento/magento2/blob/2.2/app/code/Magento/Ui/view/base/web/js/form/element/single-checkbox-toggle-notice.js

Example of usage:
I did not find a direct using of the CheckboxToggleNotice component in the Magento, but this component marked as @api.
Also, we noticed that the https://github.com/magento/magento2/blob/2.3/app/code/Magento/SalesRule/view/adminhtml/web/js/form/element/apply_to_shipping.js component inherits the CheckboxToggleNotice and the example of using the applyToShip component is here: https://github.com/magento/magento2/blob/2.3/app/code/Magento/SalesRule/view/adminhtml/ui_component/sales_rule_form.xml#L455-L481

whatsnew
Added a new topic about using the [CheckboxToggleNotice](https://devdocs/.magento.com/guides/v2.3/ui_comp_guide/components/ui-checkbox-toggle-notice.html) UI component.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.2/ui_comp_guide/components/ui-checkbox-toggle-notice.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+The CheckboxToggleNotice component implements the [`Checkbox`]({{ page.baseurl }}/ui_comp_guide/components/ui-checkbox.html) component with the ability to switch notice according to the selected option.
+
+## Configuration options
+
+| Option | Description | Type | Default |
+| --- | --- | --- | --- |
+| `component` | The path to the component’s JS constructor in terms of RequireJS. | String | `Magento_Ui/js/form/element/single-checkbox-toggle-notice` |
+| `notices` | The array of notices that should be associated by key with selected option value. | Array | `[]` |
+| `tracks`.`notice` | Flags that enable tracking for the `notice` option. | Boolean | `true` |
```
Comment (rogyar) [inline src/guides/v2.2/ui_comp_guide/components/ui-checkbox-toggle-notice.md]: Hi @serhiyzhovnir. Could you provide a bit more information information about this option? With the current explanation, a reader might still have questions regarding purpose of this option. 

Thank you!

**Comment by serhiyzhovnir** [comment]
Hi @rogyar 
I've added the detailed explanation for the `tracks`.`notice` option.
Please, check it again.
Thank you!

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**